src/coreclr/gc/unix/configure.cmake

```diff
@@ -9,6 +9,11 @@ include(CheckTypeSize)
 include(CheckLibraryExists)
 
 check_include_files(sys/time.h HAVE_SYS_TIME_H)
+
+if (CLR_CMAKE_TARGET_WASI)
+  set(CMAKE_REQUIRED_DEFINITIONS "-D_WASI_EMULATED_MMAN")
```

Comment:
Yes, its the runtime.  Looks like we need both.  The one you highlight here looks like it is so that in the configure phase it finds a `sys/mman.h` for `check_include_files(sys/mman.h HAVE_SYS_MMAN_H)`  If it doesn't then it fails with 
```
e:\github\wasi-sdk\\bin\clang++.exe --target=wasm32-wasi-threads -DBUILDENV_DEBUG=1 -DDEBUG -DFEATURE_64BIT_ALIGNMENT -DFEATURE_64BIT_ALIGNMENT=1 -DFEATURE_BASICFREEZE -DFEATURE_CACHED_INTERFACE_
  DISPATCH -DFEATURE_CONSERVATIVE_GC -DFEATURE_CUSTOM_IMPORTS -DFEATURE_DYNAMIC_CODE -DFEATURE_GC_STRESS -DFEATURE_NATIVEAOT -DFEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP -DHOST_WASM=1 -DNATIVEAOT
   -DNO_UI_ASSERT -DPAL_STDCPP_COMPAT -DTARGET_32BIT -DTARGET_UNIX -DTARGET_WASI -DTARGET_WASM -DURTBLDENV_FRIENDLY=Debug -DUSE_PORTABLE_HELPERS -DVERIFY_HEAP -D_DBG -D_DEBUG -D_FILE_OFFSET_BITS=64
   -D_LIB -D_LIBUNWIND_DISABLE_ZERO_COST_APIS=1 -D_LIBUNWIND_IS_NATIVE_ONLY -D_WASI_EMULATED_GETPID -D_WASI_EMULATED_MMAN -D_WASI_EMULATED_PROCESS_CLOCKS -D_WASI_EMULATED_SIGNAL -IE:/GitHub/runtime
  lab/src/native -IE:/GitHub/runtimelab/src/coreclr/pal/prebuilt/inc -IE:/GitHub/runtimelab/artifacts/obj -IE:/GitHub/runtimelab/src/coreclr/nativeaot/Runtime/inc -IE:/GitHub/runtimelab/src/coreclr
  /nativeaot/Runtime/. -IE:/GitHub/runtimelab/src/coreclr/nativeaot/Runtime/../../gc -IE:/GitHub/runtimelab/src/coreclr/nativeaot/Runtime/../../gc/env -IE:/GitHub/runtimelab/src/coreclr/nativeaot/R
  untime/unix -IE:/GitHub/runtimelab/src/coreclr/nativeaot/Runtime/../../pal/inc/rt -IE:/GitHub/runtimelab/src/coreclr/nativeaot/Runtime/wasm -IE:/GitHub/runtimelab/artifacts/obj/coreclr/wasi.wasm.
  Debug/nativeaot/Runtime -pthread -g -fms-extensions -Wno-unused-variable -Wno-unused-value -Wno-unused-function -Wno-tautological-compare -Wno-invalid-offsetof -Wno-deprecated-declarations -Wno-u
  nsafe-buffer-usage -Wno-unused-macros -std=gnu++11 -MD -MT nativeaot/Runtime/Portable/CMakeFiles/PortableRuntime.dir/__/__/__/gc/unix/gcenv.unix.cpp.obj -MF nativeaot\Runtime\Portable\CMakeFiles\
  PortableRuntime.dir\__\__\__\gc\unix\gcenv.unix.cpp.obj.d -o nativeaot/Runtime/Portable/CMakeFiles/PortableRuntime.dir/__/__/__/gc/unix/gcenv.unix.cpp.obj -c E:/GitHub/runtimelab/src/coreclr/gc/u
  nix/gcenv.unix.cpp
  E:/GitHub/runtimelab/src/coreclr/gc/unix/gcenv.unix.cpp:53:3: error: "sys/mman.h required by GC PAL"
   #error "sys/mman.h required by GC PAL"
```
Even though the compilation includes `-D_WASI_EMULATED_MMAN`